apps/allocations/app/components/App/PeriodDetails.js

```diff
@@ -0,0 +1,138 @@
+import React, { useState } from 'react'
+import PropTypes from 'prop-types'
+import { useAragonApi } from '../../api-react'
+import { Box, GU, Text, textStyle, useTheme } from '@aragon/ui'
+import throttle from 'lodash.throttle'
+import BigNumber from 'bignumber.js'
+import usePeriod from '../../hooks/usePeriod'
+import { ETHER_TOKEN_FAKE_ADDRESS, isTokenVerified } from '../../../../../shared/utils/token-utils'
+
+const MS_PER_DAY = 24 * 60 * 60 * 1000
+const MONTHS = [ 'Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec' ]
+const CONVERT_API_BASE = 'https://min-api.cryptocompare.com/data'
+const CONVERT_THROTTLE_TIME = 5000
+
+const convertApiUrl = symbols =>
+  `${CONVERT_API_BASE}/price?fsym=USD&tsyms=${symbols.join(',')}`
+
+const processTime = ms => {
+  const days = Math.round(ms/MS_PER_DAY)
+  if(days%7 === 0){
+    const weeks = days/7
+    return [ String(weeks), `week${weeks === 1 ? '' : 's'}` ]
+  }
+  return [ String(days), `day${days === 1 ? '' : 's'}` ]
+}
+
+const getBudgetTotals = (budgets, rates) => {
+  if(budgets.length && rates){
+    const convertedBudgets = budgets.map(budget => {
+      const rate = rates[budget.token.symbol]
+      return {
+        amount: rate ? (new BigNumber(budget.amount)).div(rate).div(10**budget.token.decimals) : new BigNumber(0),
+        remaining: rate ? (new BigNumber(budget.remaining)).div(rate).div(10**budget.token.decimals) : new BigNumber(0)
```

Comment:
Hmm, if we have rates but not rates for _this_ token, then we'll show some pretty disappointing statistics

Reply:
I'm not sure what to do here. Either we show a bunch of values denominated in their native tokens or we convert the values to USD. But if those tokens have no market value, then there is nothing to show.

Reply:
Yeah, I think this is the best we can do